std/protocols/lookup.asm

```diff
@@ -100,4 +111,21 @@ let lookup: expr[], Fp2<expr>, Fp2<expr>, Constr, expr -> () = constr |acc, alph
     is_first * acc_1 = 0;
     is_first * acc_2 = 0;
     constrain_eq_ext(update_expr, from_base(0));
+
+    // In the extension field, we need a prover function for the accumulator.
+    if needs_extension() {
+        // TODO: Helper columns, because we can't access the previous row in hints
```

Comment:
Nice, I think all we need to fix this is implement `provide_value` for `i + 1`, right?

Reply:
Yes!